lib/gtfs/data.ex

```diff
@@ -66,20 +50,29 @@ defmodule Gtfs.Data do
     ])
   end
 
+  @spec get_trip(t(), Trip.id()) :: Trip.t() | nil
+  def get_trip(%__MODULE__{trips: trips}, trip_id), do: trips[trip_id]
+
   @spec parse_files(files()) :: t()
   def parse_files(files) do
     bus_routes = Csv.parse(files["routes.txt"], &Route.bus_route_row?/1, &Route.from_csv_row/1)
     bus_route_ids = MapSet.new(bus_routes, & &1.id)
 
     bus_trips = bus_trips(files["trips.txt"], bus_route_ids)
     bus_trip_ids = MapSet.new(bus_trips, & &1.id)
+    bus_trip_stop_times = bus_trip_stop_times(files["stop_times.txt"], bus_trip_ids)
+
+    bus_trips_by_id =
```

Comment:
Maybe `bus_trips_by_id ` could be refactored into a function? It seems like several of the preceding lines are now only internal implementation details for production that result.

Reply:
Good idea. That looks much better.